Feature: Implement Custom Logger Utility

### Feature: Implement Custom Logger Utility  

#### Overview  
We need a custom `Logger` utility class to standardize logging across the project. The logger should support different log levels (`error`, `warn`, `info`), configurable minimum log levels, and structured JSON output for better readability and debugging.  

This feature will ensure that all logs in the project follow a consistent format and can be easily filtered when debugging or monitoring.  

---

####  Requirements  
- Implement a `Logger` class that:  
  - Accepts a **minimum log level** (default = `info`).  
  - Logs messages with timestamps and a standardized prefix (`[LEVEL] timestamp:`).  
  - Outputs logs in structured **JSON format** containing:  
    - `eventType` (string)  
    - `timestamp` (auto-generated)  
    - Any other custom fields passed in `LogData`.  
  - Gracefully handles errors when stringifying objects.  
- Support the following methods:  
  - `error(data: LogData)`  
  - `warn(data: LogData)`  
  - `info(data: LogData)`  

---

#### File Structure  
- Location: `utils/logger.ts`  
- Types:  
  - `ILogger` and `LoggerData` should be defined in `types/logger.types.ts`.  
  - `LogLevel` = `'error' | 'warn' | 'info'`.  
  - `LogData` = `{ eventType: string; [key: string]: unknown }`.  

---

####  Acceptance Criteria  
- Logging below the set `minLevel` is ignored.  
- Logs appear with proper prefix and ISO timestamp.  
- Log data is output as JSON.  
- Errors during logging are caught and handled without crashing the app.  
- Example usage produces structured and readable logs.  

---

####  Example Output  
```json
[ERROR] 2025-09-13T16:20:00.000Z: DB_ERROR
{
"eventType": "DB_ERROR",
"timestamp": "2025-09-13T16:20:00.000Z",
"message": "Connection failed"
}
```

---

#### Contribution Notes
- Please review our [Contribution Guidelines](https://github.com/0xdevcollins/boundless/blob/main/CONTRIBUTING.md)
 before getting started.
- Join our Telegram Channel [@boundlessfi](https://t.me/boundlessfi)


####  Notes  
- Make sure it works in both **Node.js** and **browser environments**.

--- a/lib/logger.ts
+++ b/lib/logger.ts
@@ -1,0 +1,59 @@
+import type { ILogger, LoggerData } from './types/logger.types'
+
+type LogLevel = LoggerData['LogLevel']
+type LogData = LoggerData['LogData']
+
+export class Logger implements ILogger {
+	private minLevel: LogLevel = 'info'
+
+	setMinLevel(level: LogLevel) {
+		this.minLevel = level
+	}
+
+	private shouldLog(level: LogLevel): boolean {
+		const levels: LogLevel[] = ['error', 'warn', 'info']
+		return levels.indexOf(level) <= levels.indexOf(this.minLevel)
+	}
+
+	private log(level: LogLevel, data: LogData) {
+		if (!this.shouldLog(level)) return
+
+		const timestamp = new Date().toISOString()
+		const logMethod = console[level] as (
+			message: string,
+			...optionalParams: unknown[]
+		) => void
+		const prefix = `[${level.toUpperCase()}] ${timestamp}:`
+
+		try {
+			const { eventType, ...rest } = data
+			const logData = {
+				eventType,
+				timestamp,
+				...rest,
+			}
+			const jsonData = JSON.stringify(logData, null, 2)
+			logMethod(prefix, eventType, '\n', jsonData)
+		} catch (error) {
+			console.error(`Error stringifying log data: ${error}`)
+			logMethod(
+				prefix,
+				data.eventType,
+				'\n',
+				'Error: Unable to stringify log data',
+			)
+		}
+	}
+
+	error(data: LogData) {
+		this.log('error', data)
+	}
+
+	warn(data: LogData) {
+		this.log('warn', data)
+	}
+
+	info(data: LogData) {
+		this.log('info', data)
+	}
+}

--- a/types/logger.types.ts
+++ b/types/logger.types.ts
@@ -1,0 +1,15 @@
+export interface LoggerData {
+	LogLevel: 'error' | 'warn' | 'info'
+
+	LogData: {
+		eventType: string
+		[key: string]: unknown
+	}
+}
+
+export interface ILogger {
+	setMinLevel: (level: LoggerData['LogLevel']) => void
+	error: (data: LoggerData['LogData']) => void
+	warn: (data: LoggerData['LogData']) => void
+	info: (data: LoggerData['LogData']) => void
+}

